tests/storages_tests/test_callbacks.py

```diff
@@ -0,0 +1,55 @@
+import optuna
+from optuna.storages import RetryFailedTrialCallback
+from optuna.trial import TrialState
+
+
+def test_retry_history() -> None:
+    # Test case 1: Trial that is not a retry should return empty list
+    storage = optuna.storages.RDBStorage("sqlite:///:memory:")
```

Comment:
How about using [optuna.testing.storages.StorageSupplier](https://github.com/optuna/optuna/blob/master/optuna/testing/storages.py) to instantiate the storage object? It can accept the extra keyword arguments such that `heartbeat_interbal`, and so on.

Reply:
Thank you so much for the review! 😃

I have implemented your suggestion to use `optuna.testing.storages.StorageSupplier` instead of directly using `RDBStorage`.